packages/@n8n/nodes-langchain/utils/httpProxyAgent.ts

```diff
@@ -17,3 +18,21 @@ export function getProxyAgent(targetUrl?: string) {
 
 	return new ProxyAgent(proxyUrl);
 }
+
+/**
+ * Returns a Node.js HTTP/HTTPS proxy agent for use with AWS SDK v3 clients.
+ * AWS SDK v3 requires Node.js http.Agent/https.Agent instances (not undici ProxyAgent).
+ *
+ * @param targetUrl - The target URL to check proxy configuration for
+ * @returns HttpsProxyAgent instance or undefined if no proxy is configured
+ */
+export function getNodeProxyAgent(targetUrl?: string) {
+	const proxyUrl = proxyFromEnv.getProxyForUrl(targetUrl ?? 'https://example.nonexistent/');
+
+	if (!proxyUrl) {
+		return undefined;
+	}
```

Comment:
There is an important quirk here about targetUrl optionality (see comment in the function above). Would be great to extract this block into separate function and keep the comment above